pkgs/development/python-modules/wxPython/4.0.nix

```diff
@@ -1,35 +1,83 @@
 { lib
+, stdenv
+, openglSupport ? true
+, libX11
+, pyopengl
 , buildPythonPackage
 , fetchPypi
 , pkgconfig
 , gtk3
 , libjpeg
 , libtiff
+, wxmac
+, darwin
 , SDL
 , gst-plugins-base
 , libnotify
 , freeglut
 , xorg
 , which
+, cairo
+, pango
+, python3
+, doxygen
+, ncurses
+, libpng
+, gstreamer
+, wxGTK30
+, python3Packages
 }:
 
+let
+wxGTK = wxGTK30.override { withGtk2 = false; withWebKit = true; };
+in
 buildPythonPackage rec {
   pname = "wxPython";
   version = "4.0.4";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "0d9ef4260cb2f3e23ed9dcf6baa905ba585ac7d631613cddc299c4c83463ae29";
+    sha256 = "0adfccscii4rqbfkqq9isv3mln5s0nlvmxnwv4zf5wxj1hkg97hd";
   };
 
-  nativeBuildInputs = [
-    pkgconfig
-  ];
+  doCheck = false;
 
-  buildInputs = [
-    gtk3 libjpeg libtiff SDL gst-plugins-base libnotify freeglut xorg.libSM
-    which
-  ];
+  nativeBuildInputs = [ pkgconfig which doxygen
+    ]
+    ++ (lib.optionals (!stdenv.isDarwin) [ wxGTK libX11 ])
+    ++ (lib.optionals stdenv.isDarwin [ wxmac darwin.apple_sdk.frameworks.Cocoa ]);
+
+  buildInputs = with python3Packages; [ gtk3 libjpeg libtiff SDL gst-plugins-base libnotify freeglut xorg.libSM ncurses
+      requests libpng gstreamer wxGTK
+      /* webkitgtk gstreamer-plugins-base */
+  ]
+    ++ (lib.optionals (!stdenv.isDarwin) [  (wxGTK.gtk) ])
```

Comment:
Didn't you unconditionally include `gtk3` just a few lines above?

Reply:
Honestly, I'm not completely sure about the differences between `wxGTK` / `wxGTK.gtk` / `gtk3`, but it takes about an hour to build so painful to figure out what is unnecessary with my limited knowledge. Is `wxGTK.gtk` automatically included with `wxGTK`? Let me know if you have any insight

Reply:
Well, wxGTK is just wxWidgets with GTK backend. 
And here is a definition of `wxGTK.gtk` https://github.com/NixOS/nixpkgs/blob/863bc6ed94895c340dd19c089b84b7e004c34198/pkgs/development/libraries/wxwidgets/3.1/default.nix#L81, so, perhaps, in your expression you want to use `wxGTK.gtk` instead of `gtk3` or `gtk2`.

Reply:
thx!